### PR TITLE
Issue #2571: added menu item My Searches in cardbrowser.

### PR DESCRIPTION
**Pull request by brunodea**
For now, it only shows a list of saved searches in the collection. Not able to add a new search from here.

### CODE REVIEW

**Comment by timrae** [comment]
Thanks @brunodea ! 
@hssm can you review this?

File: AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.java [hunk 1]
```diff
@@ -262,6 +264,22 @@ public boolean onSelection(MaterialDialog materialDialog, View view, int which,
...
 
+    private MaterialDialog.ListCallback mMySearchDialogListener =
+            new MaterialDialog.ListCallback() {
+        @Override
+        public void onSelection(MaterialDialog materialDialog, View view, int which,
+                                   CharSequence text) {
+            JSONObject savedFiltersObj = getCol().getConf().optJSONObject("savedFilters");
+            if (savedFiltersObj != null) {
+                String searchTerms = savedFiltersObj.optString(text.toString());
+                mSearchTerms = searchTerms != null ? searchTerms : "";
```
Comment (hssm) [inline AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.java]: `optString` will return the empty string if the key doesn't exist. Checking for null here is redundant.

File: AnkiDroid/src/main/res/values/07-cardbrowser.xml [hunk 2]
```diff
@@ -37,6 +37,10 @@
     <string name="card_browser_show_suspended">Filter suspended</string>
     <string name="card_browser_search_by_tag">Filter by tag</string>
     <string name="card_browser_tags_shown">Tags: %1$s</string>
+    <string name="card_browser_list_my_searches">My searches</string>
+    <string name="card_browser_list_my_searches_title">Choose a saved search</string>
+    <string name="card_browser_list_my_searches_empty_title">No saved searches</string>
+    <string name="card_browser_list_my_searches_empty">If you want to save searches you should use the desktop client.</string>
```
Comment (hssm) [inline AnkiDroid/src/main/res/values/07-cardbrowser.xml]: Change this to `New search terms can only be created on the desktop client.`

**Comment by hssm** [comment]
Great work! Thanks for doing this.

Two additions that would be nice.
1 - Sort the saved searches list like on the desktop client so we get the same order.
2 - After selecting a search, it should show the query in the search view. Currently you can't be certain what the query contained because it shows the results but the query doesn't display.

If would be great if you could make those changes as well.

**Comment by brunodea** [comment]
No problem, 2) is easy, but 1) I'll have to take a look in the Anki code.

**Comment by hssm** [comment]
I thought it would be the other way around :D

You should be able to do 1 easily if you change `CharSequence[] savedFilters` to `List<String> savedFilters` and then do `Collections.sort(savedFilters)`. You can pass in String ArrayLists through Bundles so I don't think it'll be a problem for building the dialog.

**Comment by brunodea** [comment]
Ah, my mistake! I saw my saved searches in the desktop and they didn't seem sorted, but now I see that they are sorted in a case sensitive way. So the order would be, for example, ```Asearch, Csearch, bsearch, fsearch, ...```.

Thanks, good thing it is easier :D

**Comment by timrae** [comment]
You can use non-case-sensitive sorting if you like. We chose to do this in the TagsDialog last time with the idea that @dae will also update the desktop version, (hopefully!) when the next major version is eventually released. I think you already made a helper function for the sorting in the tags dialog @brunodea...?

**Comment by brunodea** [comment]
I forgot about that, thanks for pointing it out. I'll take a look!

**Comment by brunodea** [comment]
@hssm, the query now appears in the searchview. Unfortunately, I didn't find a way to display the searchview, the user needs to click it to see the query..

**Comment by timrae** [comment]
This doesn't work?
http://stackoverflow.com/questions/14235677/how-do-i-open-the-searchview-programmatically

**Comment by brunodea** [comment]
@timrae, I don't know if I'm doing something wrong, but it doesn't. I've tried ```mSearchView.setIconified(false)``` and ```mSearchView.requestFocus()``` with no success.

**Comment by timrae** [comment]
How about the second answer `expandActionView()`?

**Comment by brunodea** [comment]
Nice, it worked! I'll push it right now.

**Comment by hssm** [comment]
It works, but unfortunately that method requires API level 14, and our minimum is 8 and `setIconified` requires 11. I didn't think it would be this troublesome...

If you can't find a good solution then you can pull out that change and I'll merge it without it. It's not the most important detail, just something nice to have.

**Comment by timrae** [comment]
Just do `CompatHelper.getSdkVersion() >= 14`

**Comment by hssm** [comment]
Oh wait, there is a `MenuItemCompat`! From the SO page, you can use this:
`MenuItemCompat.expandActionView(mSearchItem);`

**Comment by brunodea** [comment]
Oh god, I didn't see that. I should pay more attention when reading. Will push again now.

**Comment by hssm** [comment]
Thanks! Appreciate the contribution.

**Pull request merged**